src/client/Client.js

```diff
@@ -365,6 +366,19 @@ class Client extends BaseClient {
       .then(data => new GuildPreview(this, data));
   }
 
+  /**
+   * Obtains the widget of a guild from Discord, available for guilds with the widget enabled.
+   * @param {GuildResolvable} guild The guild to fetch the widget for
+   * @returns {Promise<Widget>}
+   */
+  async fetchWidget(guild) {
+    const id = this.guilds.resolveID(guild);
+    if (!id) throw new TypeError('INVALID_TYPE', 'guild', 'GuildResolvable');
+    const data = await this.api.guilds(id, 'widget.json').get();
+    if (data.code) throw new Error('WIDGET_DISABLED');
```

Comment:
This will never evaluate to true, since a disabled widget won't result in a response with a successful response code.
```suggestion
```

Reply:
I don't think so, I have tried last time and the error was triggered, I'll retry it and I'll update this message later